app/tests/heartbeat_controller_desktop_test.cc

```diff
@@ -336,5 +342,289 @@ TEST_F(HeartbeatControllerDesktopTest,
   }
 }
 
-}  // namespace
+TEST_F(HeartbeatControllerDesktopTest, EncodeAndDecode) {
+  std::string original_str = R"json({
+      "heartbeats": [
+        {
+          agent: "test-agent",
+          dates: ["2015-02-03"]
+        }
+      ],
+      "version":"2"
+    })json";
+  std::string encoded = controller_.CompressAndEncode(original_str);
+  std::string decoded = controller_.DecodeAndDecompress(encoded);
+  std::string expected_encoded_payload =
+      "H4sIAAAAAAAC_6vmUgADpYzUxKKSpNTEkmIlK4VoqKiCQjWcpaCQmJ6aV2KloFSSWlyiC-"
+      "Yo6SBJpyQCJYB6lYwMDE11DYx0DYyVYuHytVBWLEyLUllqUXFmfp6SlZKRElisFgDch9hUjQ"
+      "AAAA";
```

Comment:
Since you are testing against the same encoded string again and again, define this as a constant.

If the code styling allows, put the entire string the same line.

This will help
1. Code search this value in the future.  This should match the one we put in the server parser test
2. Avoid code formatter to cut this into different chunks, which make it pretty hard to tell if they are the same.

Reply:
I tweaked EncodeAndDecode so that it and CreatePayloadString can generate the same encoded payload and made it a constant. This constant should match the server parser test.

For GetExpectedHeartbeatPayload, it requires an actual user-agent which is of the form agent/version and so I don't have any way to set the user agent to "test-agent". So it has a different encoded string.